vcf.c

```diff
@@ -2901,6 +2901,22 @@ int vcf_parse(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v)
     return ret;
 }
 
+int vcf_open_mode(char *mode, const char *fn, const char *format)
+{
+    if (format == NULL) {
+        // Try to pick a format based on the filename extension
+        char extension[7];
+        if (find_file_extension(fn, extension) < 0) return -1;
+        return vcf_open_mode(mode, fn, extension);
+    }
+    else if (strcasecmp(format, "bcf") == 0) strcpy(mode, "b");
+    else if (strcasecmp(format, "vcf") == 0) strcpy(mode, "");
+    else if (strcasecmp(format, "vcf.gz") == 0 || strcasecmp(format, "vcf.bgz")) strcpy(mode, "z");
```

Comment:
Missing `== 0` for `vcf.bgz` here, which means the corresponding test in `test_open_format()` succeeds when it shouldn't due to missing `.bgz` detection in `find_file_extension()`.